localization/ndt_scan_matcher/src/ndt_scan_matcher_core.cpp

```diff
@@ -141,6 +141,21 @@ NDTScanMatcher::NDTScanMatcher()
   initial_pose_distance_tolerance_m_ =
     this->declare_parameter<double>("initial_pose_distance_tolerance_m");
 
+  use_cov_estimation_ = this->declare_parameter<bool>("use_covariance_estimation");
+  if (use_cov_estimation_) {
+    std::vector<double> initial_pose_offset_model_x =
+      this->declare_parameter<std::vector<double>>("initial_pose_offset_model_x");
+    std::vector<double> initial_pose_offset_model_y =
+      this->declare_parameter<std::vector<double>>("initial_pose_offset_model_y");
+
+    if (initial_pose_offset_model_x.size() == initial_pose_offset_model_y.size()) {
+      initial_pose_offset_model_ =
+        create_initial_pose_offset_model(initial_pose_offset_model_x, initial_pose_offset_model_y);
+    } else {
+      use_cov_estimation_ = false;
```

Comment:
I feel it would be better to make it an explicit error here than to make it false.
I am not sure if the node should be shut down, but I wanted at least a message to the log.